### PR TITLE
Added support for remembering rows per page

### PR DESCRIPTION
**Pull request by ganning127**
localStorage keys (from EnvironmentContext.js)
```
groupsPerPageKey: "zmt-perpage-groups",
resourcesPageKey: "zmt-perpage-resources",
serversPageKey: "zmt-perpage-servers",
usersPageKey: "zmt-perpage-users",
defaultAmount: 25,
```

happy to do a live demo of this. happy to discuss default amounts / naming

### CODE REVIEW

File: src/contexts/EnvironmentContext.js [hunk 1]
```diff
@@ -38,6 +38,12 @@ export const EnvironmentProvider = ({ children }) => {
             usersTitle: "Users",
             zonesTitle: "Zones",
 
+            groupsPerPageKey: "zmt-perpage-groups",
+            resourcesPageKey: "zmt-perpage-resources",
+            serversPageKey: "zmt-perpage-servers",
+            usersPageKey: "zmt-perpage-users",
+            defaultAmount: 25,
```
Comment (trel) [inline src/contexts/EnvironmentContext.js:45]: need a more specific name here... perhaps... `defaultPageSize` or `defaultItemsPerPage`

File: src/contexts/ServerContext.js [hunk 2]
```diff
@@ -518,11 +521,24 @@ export const ServerProvider = ({ children }) => {
...
+
+        for (let key of pageKeys) {
+            if (!localStorage.getItem(environment[key])) {
+                localStorage.setItem(environment[key], environment.defaultAmount);
+            }
+        }
+
+        const rescPerPage = parseInt(localStorage.getItem(environment[pageKeys[0]]));
+        const usersPerPage = parseInt(localStorage.getItem(environment[pageKeys[1]]));
+        const groupsPerPage = parseInt(localStorage.getItem(environment[pageKeys[2]]));
```
Comment (trel) [inline src/contexts/ServerContext.js:535]: are these in the correct order?   you have [groups, resources, users] in alpha order on line 525.

File: src/views/Server.js [hunk 3]
```diff
@@ -46,14 +46,29 @@ export const Server = () => {
...
     const [currServer, setCurrServer] = useState();
     const [order, setOrder] = useState("asc");
     const [orderBy, setOrderBy] = useState("role");
 
+    useEffect(() => {
+        // runs on initial render
+        const serversPerPage = localStorage.getItem(serversPageKey);
+        
+        if (!serversPerPage) {
+            localStorage.setItem(serversPageKey, environment.defaultAmount); // TODO: extract default value to another variable / file
```
Comment (trel) [inline src/views/Server.js:63]: think the TODO is done... just... need a better name (see above)

File: src/contexts/ServerContext.js (hunk 2 above)
Comment (ganning127) [reply src/contexts/ServerContext.js:535]: Ohhh, great catch. Was writing this code on a flight, must've slipped my eye 🙃

File: src/contexts/EnvironmentContext.js (hunk 1 above)
Comment (ganning127) [reply src/contexts/EnvironmentContext.js:45]: i really like `defaultItemsPerPage`. changed to that

Comment (trel) [reply src/contexts/EnvironmentContext.js:45]: bueno

**Comment by ganning127** [comment]
just rebased onto main to resolve conflicts

**Comment by trel** [comment]
squash em.  no pounds.

File: src/contexts/ServerContext.js [hunk 4]
```diff
@@ -528,11 +531,25 @@ export const ServerProvider = ({ children }) => {
...
+        
+
         !isLoadingZones && loadZones();
         !isLoadingZoneContext && loadServers();
-        !isLoadingUserContext && loadUsers(0, 10, '', 'asc', 'USER_NAME');
-        !isLoadingGroupContext && loadGroups(0, 10, '', 'asc', 'USER_NAME');
-        !isLoadingRescContext && loadResources(0, 0, '', 'asc', 'RESC_NAME');
+        !isLoadingUserContext && loadUsers(0, usersPerPage, '', 'asc', 'USER_NAME');
+        !isLoadingGroupContext && loadGroups(0, groupsPerPage, '', 'asc', 'USER_NAME');
+        !isLoadingRescContext && loadResources(0, rescPerPage, '', 'asc', 'RESC_NAME');
```
Comment (trel) [inline src/contexts/ServerContext.js:552]: let's keep these in the same order as above.   no functional change, just aesthetic.

Comment (ganning127) [reply src/contexts/ServerContext.js:552]: yep, i like the consistency better if its the same order as well. changed to

```js
!isLoadingGroupContext && loadGroups(0, groupsPerPage, '', 'asc', 'USER_NAME');
!isLoadingRescContext && loadResources(0, rescPerPage, '', 'asc', 'RESC_NAME');
!isLoadingUserContext && loadUsers(0, usersPerPage, '', 'asc', 'USER_NAME');
```

Comment (trel) [reply src/contexts/ServerContext.js:552]: perfect

**Comment by ganning127** [comment]
squashed down to one commit!

**Comment by ganning127** [comment]
yep, just did testing, works great. happy to do a live demo of this if you want. will add #

**Pull request merged**